packages/imperative/src/config/src/ProfileInfo.ts

```diff
@@ -1348,11 +1351,29 @@ export class ProfileInfo {
         const schemaPath = url.fileURLToPath(schemaUri);
 
         // if profile type schema has changed or if it doesn't exist on-disk, rebuild schema and write to disk
-        if (!sameSchemaExists && fs.existsSync(schemaPath)) {
+        if (versionChanged || !sameSchemaExists && fs.existsSync(schemaPath)) {
             jsonfile.writeFileSync(schemaPath, this.buildSchema([], layerToUpdate), { spaces: 4 });
         }
     }
 
+    /**
+     * This helper function removes all command-related properties from the given schema properties object and returns it.
+     * This is so we can easily compare schemas from disk with those that are registered with type ICommandProfileSchema.
+     * It's also been added to avoid a breaking change (as we currently allow ICommandProfileSchema objects to be registered).
+     * @param obj The properties object from the schema
+     * @returns The properties object, but with all of the command-related properties removed
+     */
+    private omitCmdPropsFromSchema(obj: Record<string, any>): Record<string, IProfileProperty> {
+        const result = lodash.omit(obj, ["optionDefinition", "optionDefinitions", "includeInTemplate"]);
```

Comment:
Is there any "constants" file that we can grab these strings from?
Otherwise, should we move these into constants to be reused elsewhere? 😋 

Or even better, should we make this array a constant of some sort that indicates the command-specific options that we are adding to the schema? 
  - Although somewhat unlikely, we can prevent having to update this in the future if we add new command-related options to the schema 😋

Reply:
I like the idea of moving these into the constants file. I've done so in commit 7ff571b 👍